### PR TITLE
Carli/1988 giant file jigsaw error

### LINKED ISSUE #1988: Giant files cause jigsaw error
When loading very big files - or even medium-big files, like the JWST NIRCAM images - I often get a "jigsaw" failure in which the screen looks like this:

image

Here's a movie showing what happens when I zoom in:
https://user-images.githubusercontent.com/143715/187045336-00b19edc-ee22-4413-864f-e51a9a39b806.mp4

In this example, the file loaded is a 2.3 GB single image (28000 x 10000 pixels)

This is with CARTA beta 3.0.0-beta3

### PR DESCRIPTION
**Pull request by crocka**
**Description**

This pr fixes #1988. Associated backend PR is [#1288](https://github.com/CARTAvis/carta-backend/pull/1288). The pendingSynchronisedTiles map in the TileService failed to clean up itself with each render, causing messy tiles to be rendered during panning and zooming. Also, syncId and tileCount are added to the RasterSyncMessage because there was not any robust method to determine whether the required tiles in a sync group have been transmitted, causing missing tiles during animation. Lastly, the pr fixed the blinking of the vector overlay during animation due to racing condition that caused image rendering before all the vector overlay tiles have been transmitted.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] e2e test passing / ~corresponding fix added~
- [x] changelog updated / ~~no changelog update needed~~
- [ ] protobuf updated to the latest dev commit / ~~no protobuf update needed~~
- [x] `BackendService` unchanged / ~~`BackendService` changed and corresponding ICD test fix added~~

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
the solution with sync_id works nicely as far as I can tell. No missing tiles after animation stopping, or during animation playback. Vector overlay is rendered properly during/after animating as well. 👍 Thanks @crocka and @pford.

**Review by veggiesaurus — APPROVED** [review]

**Review by veggiesaurus — APPROVED** [review]

File: src/services/TileService.ts [hunk 1]
```diff
@@ -399,10 +401,17 @@ export class TileService {
...
+            this.syncIdTileCountMap.set(syncMessage.syncId, syncMessage.tileCount);
+            this.syncIdMap.set(syncMessage.syncId, false);
+            if (this.pendingDecompressions.has(key)) {
+                this.pendingDecompressions.get(key).set(syncMessage.syncId, new Map<number, boolean>());
+            } else {
+                this.pendingDecompressions.set(key, new Map<number, Map<number, boolean>>().set(syncMessage.syncId, new Map<number, boolean>()));
+            }
         } else {
-            // mark the channel as complete
+            // mark the channel and as complete
```
Comment (YuHsuan-Hwang) [inline src/services/TileService.ts]: Typo?

**Review by YuHsuan-Hwang — APPROVED** [review]
Nice work. 👍

File: src/services/TileService.ts (hunk 1 above)
Comment (crocka) [reply src/services/TileService.ts]: Fixed

**Comment by YuHsuan-Hwang** [comment]
Ready for merge. @crocka Please update the change log, thanks.
I wonder if this PR also fixes missing tile related issues #2026, #954, or #794?

**Comment by crocka** [comment]
I believe that it also resolved issue #2026, but further testing is required to confirm that. @YuHsuan-Hwang

**Comment by YuHsuan-Hwang** [comment]
Let's first close 1988 for now then.

**Pull request merged**